### PR TITLE
Add linux/386 follow-up

### PR DESCRIPTION
**Pull request by crazy-max**
Follow-up #4 with emulator if you want to cherry pick @tiborvass

### CODE REVIEW

File: scripts/cross.sh [hunk 1]
```diff
@@ -68,6 +68,11 @@ case "$TARGETARCH" in
   DPKG_ARCH="s390x"
   PKG_PREFIX="s390x-linux-gnu"
   ;;
+"i386")
```
Comment (tonistiigi) [inline scripts/cross.sh:71]: `GOARCH` is `386`. In follow up we should validate binaries architectures. Atm I think no 32bit binaries were built.

File: scripts/cross.sh [hunk 2]
```diff
@@ -68,6 +68,11 @@ case "$TARGETARCH" in
...
+  OUT_ARCH="i386"
+  DPKG_ARCH="i386"
+  PKG_PREFIX="i386-linux-gnu"
```
Comment (tonistiigi) [inline scripts/cross.sh:74]: `i686-linux-gnu`

File: scripts/cross.sh (hunk 1 above)
Comment (crazy-max) [reply scripts/cross.sh]: > Atm I think no 32bit binaries were built.

Yes you're right

**Comment by crazy-max** [comment]
@tonistiigi I encountered the following stack with gcc-cross on i686:

```
#137 [buildkit-helper-all linux/amd64 build 1/2] RUN --mount=target=.,from=sr...
#137 1252.   LINK    aarch64-linux-user/qemu-aarch64
#137 1255. /usr/lib/gcc-cross/i686-linux-gnu/8/../../../../i686-linux-gnu/bin/ld: /usr/lib/i386-linux-gnu/libglib-2.0.a(gutils.c.o): in function `g_get_user_database_entry':
#137 1255. (.text+0x29c): warning: Using 'getpwuid' in statically linked applications requires at runtime the shared libraries from the glibc version used for linking
#137 1255. /usr/lib/gcc-cross/i686-linux-gnu/8/../../../../i686-linux-gnu/bin/ld: (.text+0xf6): warning: Using 'getpwnam_r' in statically linked applications requires at runtime the shared libraries from the glibc version used for linking
#137 1255. /usr/lib/gcc-cross/i686-linux-gnu/8/../../../../i686-linux-gnu/bin/ld: (.text+0x135): warning: Using 'getpwuid_r' in statically linked applications requires at runtime the shared libraries from the glibc version used for linking
#137 1255. /usr/lib/gcc-cross/i686-linux-gnu/8/../../../../i686-linux-gnu/bin/ld: /usr/lib/gcc-cross/i686-linux-gnu/8/../../../../i686-linux-gnu/lib/../lib/libc.a(memset_chk-nonshared.o): unsupported non-PIC call to IFUNC `memset'
#137 1255. /usr/lib/gcc-cross/i686-linux-gnu/8/../../../../i686-linux-gnu/bin/ld: final link failed: nonrepresentable section on output
#137 1255. collect2: error: ld returned 1 exit status
#137 1255. make[1]: *** [Makefile:208: qemu-aarch64] Error 1
#137 1255. make: *** [Makefile:527: aarch64-linux-user/all] Error 2
#137 ERROR: executor failed running [/bin/sh -c TARGETPLATFORM=${TARGETPLATFORM} configure_qemu.sh &&   make -j "$(getconf _NPROCESSORS_ONLN)" &&   make install]: exit code: 2
```

I have disabled PIE in the meantime to fix this issue.

**Review by tonistiigi — APPROVED** [review]

**Pull request merged**